service/cloudconfig/cloud_config.go

```diff
@@ -0,0 +1,103 @@
+package cloudconfig
+
+import (
+	"github.com/giantswarm/certificatetpr"
+	clustertprspec "github.com/giantswarm/clustertpr/spec"
+	cloudconfig "github.com/giantswarm/k8scloudconfig"
+	"github.com/giantswarm/kvmtpr"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+)
+
+const (
+	FileOwner      = "root:root"
+	FilePermission = 0700
+)
+
+// Config represents the configuration used to create a cloud config service.
+type Config struct {
+	// Dependencies.
+	Logger micrologger.Logger
+}
+
+// DefaultConfig provides a default configuration to create a new cloud config
+// service by best effort.
+func DefaultConfig() Config {
+	return Config{
+		// Dependencies.
+		Logger: nil,
+	}
+}
+
+// CloudConfig implements the cloud config service interface.
+type CloudConfig struct {
+	// Dependencies.
+	logger micrologger.Logger
+}
+
+// New creates a new configured cloud config service.
+func New(config Config) (*CloudConfig, error) {
```

Comment:
This is a new cloud config service to manage the version magic. I like that this is separated but I do not like the implementation. But maybe this version bullshit is just what it is.